src/renderers/dom/shared/DOMProperty.js

```diff
@@ -208,22 +219,51 @@ var DOMProperty = {
   getPossibleStandardName: __DEV__ ? {autofocus: 'autoFocus'} : null,
 
   /**
-   * All of the isCustomAttribute() functions that have been injected.
-   */
-  _isCustomAttributeFunctions: [],
-
-  /**
-   * Checks whether a property name is a custom attribute.
+   * Checks whether a property name is a writeable attribute.
    * @method
    */
-  isCustomAttribute: function(attributeName) {
-    for (var i = 0; i < DOMProperty._isCustomAttributeFunctions.length; i++) {
-      var isCustomAttributeFn = DOMProperty._isCustomAttributeFunctions[i];
-      if (isCustomAttributeFn(attributeName)) {
+  shouldSetAttribute: function(name, value) {
+    if (
+      DOMProperty.isReservedProp(name) ||
+      DOMProperty.aliases.hasOwnProperty(name)
+    ) {
+      return false;
+    }
+
+    if (DOMProperty.properties.hasOwnProperty(name)) {
+      return true;
+    }
+
+    if (value === null) {
+      return true;
+    }
+
+    if (name.toLowerCase() !== name) {
+      return false;
+    }
```

Comment:
TLDR: I don't love this. I think we should just allow any casing for attributes outside the whitelist so that we can remove most of the white list and gracefully handle SVG's case sensitivity (if that's really a thing).

This is necessary if we don't want to allow things like `data-fooBar` or `myCustomAttribute` to show up in the DOM. 

[MDN suggests that SVG attributes are case sensitive](https://developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Introduction). That's a huge bummer.... maybe we really can't remove SVG attributes from a special list. 

Since HTML attributes aren't case sensitive, I'd rather just set attributes with the casing they are provided as props, and let the DOM downcase them for us (which it will do anyway). We'd allow custom casing (what if there's a new SVG property that comes out that we don't support?)